### PR TITLE
Fix OperatorApply Bug

### PR DESCRIPTION
**Pull request by jeremylt**
Fixes the bug identified in Ratel. See discussion here: https://gitlab.com/micromorph/ratel/-/merge_requests/464

### CODE REVIEW

File: interface/ceed-operator.c [hunk 1]
```diff
@@ -1513,6 +1494,21 @@ int CeedOperatorApply(CeedOperator op, CeedVector in, CeedVector out, CeedReques
...
+    } else {
+      // Zero all output vectors
+      CeedQFunction qf = op->qf;
+      for (CeedInt i = 0; i < qf->num_output_fields; i++) {
+        CeedVector vec = op->output_fields[i]->vec;
+        if (vec == CEED_VECTOR_ACTIVE) vec = out;
+        if (vec != CEED_VECTOR_NONE) CeedCall(CeedVectorSetValue(vec, 0.0));
+      }
+      // Apply
+      if (op->num_elem) CeedCall(op->ApplyAdd(op, in, out, request));
```
Comment (jrwrigh) [inline interface/ceed-operator.c:1510]: Do we want to catch if we get to this point and `op->num_elem` is NULL? Seems weird that we might get this far, not actually apply the operator, but not return an error or something.

File: interface/ceed-operator.c [hunk 2]
```diff
@@ -1552,6 +1545,9 @@ int CeedOperatorApplyAdd(CeedOperator op, CeedVector in, CeedVector out, CeedReq
         CeedCall(CeedOperatorApplyAdd(sub_operators[i], in, out, request));
       }
     }
+  } else if (op->num_elem) {
+    // Standard Operator
+    CeedCall(op->ApplyAdd(op, in, out, request));
```
Comment (jrwrigh) [inline interface/ceed-operator.c:1550]: Similar here. Not sure the likelihood of both `op->is_composite` and `op->num_elem` being NULL though.

**Review by jrwrigh — APPROVED** [review]

File: interface/ceed-operator.c (hunk 1 above)
Comment (jeremylt) [reply interface/ceed-operator.c:1510]: `op->num_elem` is set to `-1` initially. Having `op->num_elem == 0` means that the operator has zero elements on the current process (think like a boundary condition operator)

Comment (jeremylt) [reply interface/ceed-operator.c:1510]: By accepting 0 elements on the current process as valid, we need a small amount of special case logic in libCEED but prevent the caller from having to wrap similar special case logic around all of their operators.

Comment (jeremylt) [reply interface/ceed-operator.c:1510]: Found it! Here's the original bugfix that added this logic: https://github.com/CEED/libCEED/pull/383

Comment (jrwrigh) [reply interface/ceed-operator.c:1510]: Digging that PR from way back. Haha.

But gotcha, makes sense.

**Pull request merged**